Automattic/wp-calypso
# Update the the link on the license activation screen to go to the correct plugin page

## Proposed Changes

* This PR does a better plugin check when choosing what wp-admin URL to link to
* This check is improved by allowing the product slug to be checked if one is passed

## Testing Instructions

* First, test WITHOUT the PR to reproduce the issue
* Obtain a test Jetpack site with both Jetpack Search and Jetpack VideoPress plugins installed 
* Start disconnected on the Jetpack site and purchase Jetpack VideoPress through My Jetpack
* You should complete the purchase and be redirected to connect your account afterwards
* Once connected, the license should automatically assign to the site you came from, and you will land on the activation thank you screen:
![Screenshot 2024-02-27 at 2 55 13 PM](https://github.com/Automattic/wp-calypso/assets/18016357/388bc70a-4de2-4469-906e-faccd6d84552)
* Notice that the "Go to dashboard" link here links to the Jetpack Search plugin page on your site.

* Now, apply this PR on your local Calypso instance
* Copy and paste the URL from the checkout thank you page in the previous set of steps and paste it into the local calypso instance (using calypso.localhost instead of wordpress.com)
* Now, the "Go to Dashboard" button should correctly link to the VideoPress plugin page

## Pre-merge Checklist
